src/markdown.c

```diff
@@ -2530,16 +2530,15 @@ sd_markdown_new(
 	if (md->cb.linebreak)
 		md->active_char['\n'] = MD_CHAR_LINEBREAK;
 
-	if (md->cb.image || md->cb.link)
-		md->active_char['['] = MD_CHAR_LINK;
-
 	md->active_char['<'] = MD_CHAR_LANGLE;
 	md->active_char['\\'] = MD_CHAR_ESCAPE;
 	md->active_char['&'] = MD_CHAR_ENTITITY;
 
 	if (extensions & MKDEXT_AUTOLINK) {
 		if (!(extensions & MKDEXT_NO_EMAIL_AUTOLINK))
 			md->active_char['@'] = MD_CHAR_AUTOLINK_EMAIL;
+		if (md->cb.image || md->cb.link)
```

Comment:
Agreed. We also don't want to skip by the link structure entirely, we just want to not render the link. Outputting `[foo](/bar)` verbatim probably isn't the way to go either.